pkg/middleware/cached.go

```diff
@@ -59,7 +62,9 @@ func (mw *CachedOptlyMiddleware) ClientCtx(next http.Handler) http.Handler {
 			return
 		}
 
-		optlyClient, err := mw.Cache.GetClient(sdkKey)
+		datafileAccessToken := r.Header.Get(OptlyDatafileAccessToken)
+
+		optlyClient, err := mw.Cache.GetClient(sdkKey, datafileAccessToken)
```

Comment:
Imagine the following scenario:

1. Agent started with configuration: `datafileAccessTokens: {"mySdkKey": ["myToken1"]}`
2. Client added to cache with key: `"mySdkKey:myToken1"`
3. Request comes in with `X-Optimizely-Sdk-Key: mySdkKey`, and **no X-Optimizely-Datafile-Access-Token**

In this scenario, I think we want the existing client to be used. But this implementation will access the cache using only the sdk key as the cache key, not find anything, and then attempt to create a new client with SDK key only. If that SDK key corresponds to an authenticated datafile, it will fail to load.

Probably we want to treat the access token in the header as a "force" or "override" configuration - ignore any existing client we have for that SDK key, and create a new one using that SDK key and that access token.

Reply:
I thought we explicitly want the composite cache key behavior. Having redundant instances of Optly in the cache is ok since they are issued and revoked independently. Forcing an override when the access token changes can result in large number of cache reloads if two independent keys, from independent services are accessing the same Agent cluster.

Reply:
Yea that's a good point about forcing an override.

My main question is: will we require the datafile access token to be included in the request header, in order to use Agent with an authenticated datafile? If the answer is yes, this implementation works, otherwise it doesn't.

Reply:
If we dont make datafile access tokens required in the request, then instead, they can be set in Agent configuration, Agent will fetch authenticated datafiles as needed, and requesters don't worry about it.